VirtualFinland.UserAPI/src/VirtualFinland.UsersAPI/Program.cs

```diff
@@ -124,12 +125,15 @@
 // Redis connection
 //
 var redisEndpoint = Environment.GetEnvironmentVariable("REDIS_ENDPOINT") ?? builder.Configuration["Redis:Endpoint"];
-ConnectionMultiplexer redis = ConnectionMultiplexer.Connect($"{redisEndpoint},abortConnect=false,connectRetry=5");
+ConnectionMultiplexer redisCluster = ConnectionMultiplexer.Connect($"{redisEndpoint},abortConnect=false,connectRetry=5");
+IDatabase redisDatabase = redisCluster.GetDatabase();
+builder.Services.AddSingleton(redisDatabase);
+builder.Services.AddSingleton<ICacheRepositoryFactory, CacheRepositoryFactory>();
```

Comment:
Kertauksen vuoksi: mikäs tämän rediksen funktio oli tässä ennen näitä lisäyksiä? Onko tuo database (cache) nyt jaettu setti edellisen implementaation kanssa?

Reply:
Lambda-funktion ajojen elinkaaren takia ajomuistin käyttö välimuistina eli cachena ei ole kovin ennustettavaa tai kontrolloitava, esim. jos haluaisi että on cache-tietueella globaali TTL, ei se oikein pelkästään lambdojen muistilla taida hoitua. TTL:n hallinta taas on tärkeää varsinkin kun käsitellään ulkopuolisia palveluja, että kuinka tiheään lähetetään niihin kyselyjä. Tuota varten välimuistihommeleille ajontilan ulkopuolinen säiliö ja tuosta tekkivalinnoista hyvinä vaihtiksina aws:llä elasticache (jossa redis) tai dynamodb. Pääasia että on optimoitu nopeaan lukemiseen ja helposti skaalautuva palikka. Rediksellä on clienttipäässä mun tuntumalla yksinkertaisempi meno kun sitä ei ole tarkoitettu niin monikäyttöiseksi kuin dynamodb. 

Elikkäs sitä redistä käytettiin ennen tätä PR:ää siihen audience guardi-palikkaan jossa kakutetaan se dataspace-apilta kysyttävä pyyntö jossa tietoja kyseisestä appiksesta, eli että ei tarvitse koko ajan varmistaa samaa audiencia vaan yhden varmistuksen voimassaolon aika on määritelty ajaksi x. Uudet cache-toteutukset käyttää samaa kantaa, mutta tuossa on mukana vartiaa ja työkalua että ei tule törmäilyä nimi-avaruuksien kanssa.